src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -470,31 +492,14 @@ private void ScanFormatSpecifier()
                             return; // premature end of string! let caller complain about unclosed interpolation
                         }
                     }
-                    else if (ch == '{')
+                    else if (ch is '{' or ':')
```

Comment:
why the `:` part?

Reply:
Because the colon is not allowed according to the spec

Reply:
i don't think we should be making any unnecessary changes.  esp. those that might break people.  The only approved change we ran by the LDM was updating curly brace matching to match what the runtime does. 

Unless the runtime also has this behavior for things like `:` and `"`, then we should not touch it.

Reply:
Okay, I'm fine to allow those - for me even '{' would be okay

Reply:
we should disallow `{` so that we can be in sync with teh runtime.  The goal here is to get us operating the same way so we're in agreemnt on the processing.  ideally we shoudl only make one change and be done here :)